File: app/src/main/java/com/verbatoria/presentation/schedule/view/adapter/viewholder/ScheduleItemViewHolder.java
package com.verbatoria.presentation.schedule.view.adapter.viewholder;

import android.view.View;

import com.cleveroad.adaptivetablelayout.ViewHolderImpl;
import com.remnev.verbatoriamini.R;

/**
 * View Holder для отображения ячейки события
 *
 * @author nikitaremnev
 */
public class ScheduleItemViewHolder extends ViewHolderImpl {

    private View mRootView;

    public ScheduleItemViewHolder(View itemView) {
        super(itemView);
        mRootView = itemView;
    }

    public void setChecked(boolean checked) {
        if (checked) {
            mRootView.setBackgroundResource(R.color.main_dark);
        } else {
            mRootView.setBackgroundResource(R.color.background);
        }
    }

}

File: app/src/main/java/com/verbatoria/ui/schedule/item/ScheduleItemViewHolder.kt
package com.verbatoria.ui.schedule.item

import android.view.View
import com.remnev.verbatoria.R
import com.verbatoria.ui.common.adaptivetablelayout.ViewHolderImpl

/**
 * @author n.remnev
 */

class ScheduleItemViewHolder(
    private val backgroundView: View
) : ViewHolderImpl(backgroundView) {

    fun setChecked(isChecked: Boolean) {
        if (isChecked) {
            backgroundView.setBackgroundResource(R.color.main)
        } else {
            backgroundView.setBackgroundResource(R.color.background)
        }
    }

}